### PR TITLE
Add CI workflow

### PR DESCRIPTION
**Pull request by h4nsu**
This PR adds a simple CI workflow performing fmt, clippy, unit tests and making sure the contract builds.

As a bonus, Cargo.toml packaga manifest is updated in preparation for crate publishing

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Cache Crates
+        uses: actions/cache@v3
+        with:
+          path: ~/.cargo
+          key: ${{ runner.os }}-contract-build-env
```
Comment (deuszx) [inline .github/workflows/unit_tests.yml:54]: This will  cache `~/.cargo`  across different Rust toolchain versions. Is that intended?

Comment (h4nsu) [reply .github/workflows/unit_tests.yml:54]: Yes, it is

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Run cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all
+
+      - name: Run clippy
```
Comment (deuszx) [inline .github/workflows/unit_tests.yml:31]: Clippy needs to download dependencies (similarly the "run unit tests" job) so we should recover cache before these steps as well.

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (deuszx) [reply .github/workflows/unit_tests.yml:54]: but why? We probably won't need more than 1 version at the time so why cache them all?

Comment (h4nsu) [reply .github/workflows/unit_tests.yml:54]: Hmm, here we are doing everything with the default stable Rust, without resorting to "toolchain versioning" we do in other repos. I'm not sure, but I think that if the `stable` toolchain gets updated "behind the curtains", this workflow will just keep working without bloating the cache...

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (h4nsu) [reply .github/workflows/unit_tests.yml:31]: It takes ~1 minute to do this whole job without caching, sounds like a bit of an overkill

Comment (deuszx) [reply .github/workflows/unit_tests.yml:31]: Ok, agreed.

File: .github/workflows/unit_tests.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Run cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all
```
Comment (Marcin-Radecki) [inline .github/workflows/unit_tests.yml:29]: I think we need to pass `--check` as well (see https://github.com/rust-lang/rustfmt#checking-style-on-a-ci-server)

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (Marcin-Radecki) [reply .github/workflows/unit_tests.yml:54]: About this cache: with GH runners it is fine to choose whatever caching mechanism as it is separated from self-hosted machines we use anyway. So as long as it improves something, we can use it. 

We had some issues in cargo cache last time, but we did not get to the bottom of the problem, so maybe it's worth to give it a second go now. I'm just curious for example, does this influence `cargo contract install`? IT should right? But the time of the build is always 19 minutes.

**Review by Marcin-Radecki — APPROVED** [review]

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (h4nsu) [reply .github/workflows/unit_tests.yml:54]: Yes, caching is here precisely for avoiding loooong cargo contract compilation time. And it seems to work:
https://github.com/Cardinal-Cryptography/PSP22/actions/runs/6810159396/job/18517919901

File: .github/workflows/unit_tests.yml (hunk 3 above)
Comment (h4nsu) [reply .github/workflows/unit_tests.yml]: done

**Pull request merged**